include/nbl/asset/CCPUMeshPacker.h

```diff
@@ -71,38 +71,142 @@ class CCPUMeshPacker final : public IMeshPacker<ICPUMeshBuffer, MDIStructType>
 	using TriangleBatch = typename base_t::TriangleBatch;
 
 public:
-	CCPUMeshPacker(const SVertexInputParams& preDefinedLayout, const MeshPackerBase::AllocationParams& allocParams, uint16_t minTriangleCountPerMDIData = 256u, uint16_t maxTriangleCountPerMDIData = 1024u)
-		:IMeshPacker<ICPUMeshBuffer, MDIStructType>(preDefinedLayout, allocParams, minTriangleCountPerMDIData, maxTriangleCountPerMDIData)
-	{}
+	struct AllocationParams
+	{
+		size_t indexBuffSupportedCnt = 1073741824ull;                  /*   2GB*/
+		size_t vertexBuffSupportedCnt = 107374182ull;                  /*   2GB assuming vertex size is 20B*/
+		size_t perInstanceVertexBuffSupportedCnt = 3355443ull;         /*  32MB assuming per instance vertex attrib size is 10B*/
+		size_t MDIDataBuffSupportedCnt = 16777216ull;                  /*  16MB assuming MDIStructType is DrawElementsIndirectCommand_t*/
+		size_t vertexBufferMinAllocSize = 32ull;
+		size_t indexBufferMinAllocSize = 256ull;
+		size_t perInstanceVertexBufferMinAllocSize = 32ull;
+		size_t MDIDataBuffMinAllocSize = 32ull;
+	};
+
+	struct ReservedAllocationMeshBuffers
+	{
+		uint32_t mdiAllocationOffset;
+		uint32_t mdiAllocationReservedSize;
+		uint32_t instanceAllocationOffset;
+		uint32_t instanceAllocationReservedSize;
+		uint32_t indexAllocationOffset;
+		uint32_t indexAllocationReservedSize;
+		uint32_t vertexAllocationOffset;
+		uint32_t vertexAllocationReservedSize;
+
+		inline bool isValid()
+		{
+			return this->mdiAllocationOffset != core::GeneralpurposeAddressAllocator<uint32_t>::invalid_address;
+		}
+	};
+
+	struct PackedMeshBuffer
+	{
+		//or output should look more like `return_type` from geometry creator?
+		//TODO: add parameters of the 
+		core::smart_refctd_ptr<ICPUBuffer> MDIDataBuffer;
+		SBufferBinding<ICPUBuffer> vertexBufferBindings[SVertexInputParams::MAX_ATTR_BUF_BINDING_COUNT] = {};
+		SBufferBinding<ICPUBuffer> indexBuffer;
+
+		SVertexInputParams vertexInputParams;
+
+		inline bool isValid()
+		{
+			return this->MDIDataBuffer->getPointer() != nullptr;
+		}
```

Comment:
both packer types share a declaration of `MDIDataBuffer`, if they had a common base, they could also commonalize this `isValid` method

Reply:
TODO